modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -357,36 +402,94 @@ private static LocalPartitionStateEnum convertState(State nodeState) {
      * known.
      */
-    private Map<TablePartitionId, Map<String, LocalPartitionState>> normalize(
-            Map<TablePartitionId, Map<String, LocalPartitionState>> result
+    private static Map<TablePartitionId, Map<String, LocalPartitionState>> normalizeLocal(
+            Map<TablePartitionId, Map<String, LocalPartitionStateMessage>> result,
+            Catalog catalog
     ) {
         return result.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, entry -> {
-            Map<String, LocalPartitionState> map = entry.getValue();
+            TablePartitionId tablePartitionId = entry.getKey();
+            Map<String, LocalPartitionStateMessage> map = entry.getValue();
 
             // noinspection OptionalGetWithoutIsPresent
-            long maxLogIndex = map.values().stream().mapToLong(LocalPartitionState::logIndex).max().getAsLong();
+            long maxLogIndex = map.values().stream().mapToLong(LocalPartitionStateMessage::logIndex).max().getAsLong();
 
             return map.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, entry2 -> {
-                LocalPartitionState state = entry2.getValue();
+                LocalPartitionStateMessage stateMsg = entry2.getValue();
 
-                if (state.state() != LocalPartitionStateEnum.HEALTHY || maxLogIndex - state.logIndex() < CATCH_UP_THRESHOLD) {
-                    return state;
+                LocalPartitionStateEnum stateEnum = stateMsg.state();
+
+                if (stateMsg.state() == HEALTHY && maxLogIndex - stateMsg.logIndex() >= CATCH_UP_THRESHOLD) {
+                    stateEnum = CATCHING_UP;
                 }
 
-                return MSG_FACTORY.localPartitionState()
-                        .state(LocalPartitionStateEnum.CATCHING_UP)
-                        .partitionId(state.partitionId())
-                        .logIndex(state.logIndex())
-                        .build();
+                CatalogTableDescriptor tableDescriptor = catalog.table(tablePartitionId.tableId());
+                return new LocalPartitionState(tableDescriptor.name(), tablePartitionId.partitionId(), stateEnum);
             }));
         }));
     }
+
+    private static Map<TablePartitionId, GlobalPartitionState> normalizeGlobal(
+            Map<TablePartitionId, Map<String, LocalPartitionState>> localResult,
+            Catalog catalog
+    ) {
+        Map<Integer, Integer> tableIdToPartitions = new HashMap<>();
+
+        Map<TablePartitionId, GlobalPartitionState> result = localResult.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> {
+                    TablePartitionId tablePartitionId = entry.getKey();
+                    Map<String, LocalPartitionState> map = entry.getValue();
+
+                    int zoneId = catalog.table(tablePartitionId.tableId()).zoneId();
```

Comment:
Is it somehow guaranteed that this node contains in its local Catalog all tables that might come from other nodes in responses?

Reply:
After a second thought - not necessarily. It's not clear what we should do with deleted tables, that might cause some issues. I'll do something, and probably create a TODO for further actions

Reply:
Yes, the code seems to be correct, but it's not too obvious. Could you please add some comments on this in the code? Otherwise it will be easy to break it in the future.